pandas/core/arrays/base.py

```diff
@@ -653,15 +655,73 @@ def copy(self, deep=False):
         raise AbstractMethodError(self)
 
     # ------------------------------------------------------------------------
-    # Block-related methods
+    # Printing
     # ------------------------------------------------------------------------
+    def __repr__(self):
+        from pandas.io.formats.printing import format_object_summary
+
+        template = (
+            u'{class_name}'
+            u'{data}\n'
+            u'Length: {length}, dtype: {dtype}'
+        )
+        # the short repr has no trailing newline, while the truncated
+        # repr does. So we include a newline in our template, and strip
+        # any trailing newlines from format_object_summary
+        data = format_object_summary(self, self._formatter(), name=False,
+                                     trailing_comma=False).rstrip()
+        class_name = u'<{}>\n'.format(self.__class__.__name__)
+        return template.format(class_name=class_name, data=data,
+                               length=len(self),
+                               dtype=self.dtype)
+
+    def _formatter(self, formatter=None):
+        # type: (Optional[ExtensionArrayFormatter]) -> Callable[[Any], str]
+        """Formatting function for scalar values.
+
+        This is used in the default '__repr__'. The formatting function
+        receives instances of your scalar type.
+
+        Parameters
+        ----------
+        formatter: GenericArrayFormatter, optional
+            The formatter this array is being rendered with. When the array
+            is being rendered inside an Index, Series, or DataFrame, a
+            formatter will be provided. So if you want your objects to
+            render differently inside a Series from on its own, checking
+            with ``formatter is None`` is an option.
```

Comment:
Are you only expected to simply do such check, to return something differently, or would you actually use some information of that formatter? 
(because for the latter, just a boolean flag would be sufficient?)

EDIT: ah, I see a couple of lines below that `formatter.formatter` is used.

Reply:
I'm actually not sure we can get in a situation where formatter will be specified to anything other than the default of None. There's a lot of indirection in this code right now :/